support-frontend/assets/pages/contributions-checkout-in-epic/ContributionsCheckout.tsx

```diff
@@ -0,0 +1,166 @@
+import type {
+	ContributionAmounts,
+	ContributionType,
+} from 'helpers/contributions';
+import { getAmount } from 'helpers/contributions';
+import type { Action } from 'pages/contributions-landing/contributionsLandingActions';
+import {
+	selectAmount,
+	updateContributionTypeAndPaymentMethod,
+	updateOtherAmount,
+} from 'pages/contributions-landing/contributionsLandingActions';
+import type { State } from 'pages/contributions-landing/contributionsLandingReducer';
+import { ContributionsCheckoutForm } from './ContributionsCheckoutForm';
+import { ContributionsCheckoutSubmitting } from './ContributionsCheckoutSubmitting';
+import { ContributionsCheckoutThankYou } from './ContributionsCheckoutThankYou';
+
+// ---- Component ---- //
+
+type Status = 'FORM' | 'SUBMITTING' | 'SUCCESS';
+
+const getStatus = (state: State): Status => {
+	if (state.page.form.paymentComplete) {
+		return 'SUCCESS';
+	} else if (state.page.form.isWaiting) {
+		return 'SUBMITTING';
+	}
+	return 'FORM';
+};
+
+const getAmounts = (state: State): ContributionAmounts => {
+	const amounts = state.common.amounts;
+
+	return {
+		ONE_OFF: {
+			...amounts.ONE_OFF,
+			amounts: amounts.ONE_OFF.amounts.slice(0, 2),
+		},
+		MONTHLY: {
+			...amounts.MONTHLY,
+			amounts: amounts.MONTHLY.amounts.slice(0, 2),
+		},
+		ANNUAL: {
+			...amounts.ANNUAL,
+			amounts: amounts.ANNUAL.amounts.slice(0, 2),
+		},
+	};
+};
+
+const mapStateToProps = (state: State) => ({
+	status: getStatus(state),
+	country: state.common.internationalisation.countryId,
+	countryGroupId: state.common.internationalisation.countryGroupId,
+	currency: state.common.internationalisation.currencyId,
+	contributionType: state.page.form.contributionType,
+	amounts: getAmounts(state),
+	selectedAmounts: state.page.form.selectedAmounts,
+	otherAmounts: state.page.form.formData.otherAmounts,
+});
+
+const mapDispatchToProps = (dispatch: ThunkDispatch<State, void, Action>) => {
+	return {
+		setSelectedContributionType: (contributionType: ContributionType) => {
+			dispatch(
+				updateContributionTypeAndPaymentMethod(contributionType, 'None'),
+			);
+		},
+		setSelectedAmount: (
+			amount: number | 'other',
+			contributionType: ContributionType,
+		) => {
+			dispatch(selectAmount(amount, contributionType));
+		},
+		setOtherAmount: (amount: string, contributionType: ContributionType) => {
+			dispatch(updateOtherAmount(amount, contributionType));
+		},
+	};
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+type ContributionsCheckoutProps = ConnectedProps<typeof connector>;
+
+function ContributionsCheckout({
+	status,
+	country,
+	countryGroupId,
+	currency,
+	contributionType,
+	amounts,
+	selectedAmounts,
+	otherAmounts,
+	setSelectedContributionType,
+	setSelectedAmount,
+	setOtherAmount,
+}: ContributionsCheckoutProps): JSX.Element {
+	const ref = useOnHeightChangeEffect((height) => {
+		window.parent.postMessage({ type: 'IFRAME_HEIGHT', height }, '*');
```

Comment:
Ideally we'd maybe lock the `targetOrigin` here down to `www.theguardian.com` but it makes it a little tricky to test locally in storybook or dcr